monai/networks/nets/autoencoder.py

```diff
@@ -16,8 +16,10 @@
 
 from monai.networks.blocks import Convolution, ResidualUnit
 from monai.networks.layers.factories import Act, Norm
+from monai.networks.nets.net_factory import NetworkFactory
 
 
+@NetworkFactory.factory_function("AutoEncoder")
```

Comment:
This decorator is meant for callable objects that return the type, so these shouldn't applied to a class. For example we have 

```python
@Dropout.factory_function("alphadropout")
def alpha_dropout_factory(_dim):
      return nn.AlphaDropout
```

The decorator is applied to a function in this case but it could be to a class to act as another factory whose `__call__` method would have to return newly created objects. That's not what `AutoEncoder` is so we need a different mechanism, either adding a function like the example or we add a new decorator to the factories that accept class types as arguments and store a callable factory function into the factory.